src/api/api.jl

```diff
@@ -16,4 +16,24 @@ include("error.jl")
 include("functions.jl") # core API ccall wrappers
 include("helpers.jl")
 
+function __init__()
+    # HDF5.API.__init__() is run before HDF5.__init__()
+
+    # From deps.jl
+    check_deps()
+
+    # Ensure this is reinitialized on using
+    libhdf5handle[] = Libdl.dlopen(libhdf5)
```

Comment:
This is needed to ensure the handle doesn't change between precompilation ?

Reply:
> In the future, we should just grab the handle from HDF5_jll.

How would that that actually reinitialize the library? It seems like you'd have to call `dlopen` here regardless